### ISSUE TITLE
build_type=None cause error while building zlib package

### ISSUE THREAD

**Issue by walterj-adsk** [conan-io/conan#7762]
I'm building a clang-cl based version of zlib library.  I'm using the recipe found here: https://github.com/conan-io/conan-center-index/tree/master/recipes/zlib

We are migrating some of our packages over to clang-cl and using the following build profile:

```
[settings]
os=Windows
os_build=Windows
compiler=clang
compiler.version=9
compiler.runtime=MT
arch=x86_64
arch_build=x86_64
[options]
[env]
CC=clang-cl  -mrtm
CXX=clang-cl  -mrtm
CFLAGS= -fms-compatibility-version=1800
CXXFLAGS= -fms-compatibility-version=1800
CONAN_CMAKE_GENERATOR=Ninja
PATH=[C:\Program Files\LLVM\bin]
[build_requires]
ninja_installer/1.8.2@bincrafters/stable

```
The build works fine but the package fails and never properly copies the lib or makes a package directory.  This is the error I get:

```
zlib/1.2.11@sugarhill/stable: Package '1cfdcda7144dcb30f51594aad8c29d106a7734a3' built
zlib/1.2.11@sugarhill/stable: Build folder c:\users\walterj\.conan\data\zlib\1.2.11\sugarhill\stable\build\1cfdcda7144dcb30f51594aad8c29d106a7734a3
zlib/1.2.11@sugarhill/stable: Generated conaninfo.txt
zlib/1.2.11@sugarhill/stable: Generated conanbuildinfo.txt
zlib/1.2.11@sugarhill/stable: Generating the package
zlib/1.2.11@sugarhill/stable: Package folder c:\users\walterj\.conan\data\zlib\1.2.11\sugarhill\stable\package\1cfdcda7144dcb30f51594aad8c29d106a7734a3
zlib/1.2.11@sugarhill/stable: Calling package()
ERROR: zlib/1.2.11@sugarhill/stable: Error in package() method, line 170
        self._rename_libraries()
while calling '_rename_libraries', line 142
        os.rename(current_lib, os.path.join(lib_path, "zlib.lib"))
        FileNotFoundError: [WinError 2] The system cannot find the file specified: 'c:\\...\.conan\\data\\zlib\\1.2.11\\sugarhill\\stable\\package\\1cfdcda7144dcb30f51594aad8c29d106a7734a3\\lib\\zlibstatic.lib' -> 'c:\\users\\...\\.conan\\data\\zlib\\1.2.11\\...\\stable\\package\\1cfdcda7144dcb30f51594aad8c29d106a7734a3\\lib\\zlib.lib'

```

### Environment Details (include every applicable attribute)
  * Windows 10 - 1909
  * Clang 9 using clang-cl:
clang version 9.0.0 (tags/RELEASE_900/final)
Target: x86_64-pc-windows-msvc
Thread model: posix
InstalledDir: C:\Program Files\LLVM\bin
  * Conan version: 1.29.2
  * Python version: 3.7

### Steps to reproduce (Include if Applicable)

This is the command that I run
```
conan create . "zlib/1.2.11@sugarhill/stable" -pr ..\..\..\..\sugarhill\re\build_profiles\clang
```

**Comment by walterj-adsk** [comment]
So I think I might have solved this issue on my own.  

```
build_type = Release | Debug
```

was missing from my profile.

**Comment by walterj-adsk** [comment]
So I think I might have solved this issue on my own.  

```
build_type = Release | Debug
```

was missing from my profile.

**Comment by memsharded** [comment]
Hi @walterj-adsk 

Yes, indeed you need to have the ``build_type`` specified.
The issue is that the default settings contain ``None`` as a possible value:

```yaml
build_type: [None, Debug, Release, RelWithDebInfo, MinSizeRel]
```
While other settings do not contain the ``None`` value, and then will complain if they don't get a value, the ``build_type`` doesn't complain, but if later the recipe is not prepared for handling the build with build_type=None, then it will fail in an uglier way.

I am thinking that we might need to remove that ``None``, I don't like it very much, and while it seems that is ther because of some user request, I cannot think of a use case. If a package doesn't require a build_type, then it can remove it from the recipe, no need to have a ``None`` value.

Thanks for reporting, this might be an improvement for Conan 2.0.

**Comment by memsharded** [comment]
Update:

- the null value in settings.yml couldn't be removed. Now it is ``build_type: [null, Debug, Release, RelWithDebInfo, MinSizeRel]`` to be real Yaml specification, but still there are users and recipes that do not define at all build_type and forcing it would be bad.
- Several checks have been introduced in the new build-helpers integrations, like the new ``from conan.tools.cmake import CMake, CMakeToolchain``, and now it gives a clear error:
```
ERROR: pkg/0.1: Error in build() method, line 40
        cmake.build()
        ConanException: build_type setting should be defined.
```

So I think this can be closed now as solved in 2.0, thanks for the feedback!
